### ISSUE TITLE
Help with proper commands to add Camera, Microphone to CFBundleidentifier ca.derivative.TouchDesigner in Mojave.

### ISSUE THREAD

**Issue by FreQRiDeR** [jacobsalmela/tccutil#55]
Help with proper commands to add Camera, Microphone to CFBundleidentifier ca.derivative.TouchDesigner in Mojave.

Tried 'sudo tccutil.py -i ca.derivative.TouchDesigner --service Camera -v ‘  and got this: 
Mac-Pro:~ me$ sudo tccutil.py -i ca.derivative.TouchDesigner --service Camera -v 
Opening Database...
Database opened.

Detected "ca.derivative.TouchDesigner" as Bundle ID.
Inserting "ca.derivative.TouchDesigner" into Database...
Committing Changes...

Closing Database...
Database closed.
Mac-Pro:~ me$ 

Mic, Cam not showing in Privacy settings. ca.derivative.TouchDesigner shows in tccutil.py -l. Sip is disabled. 

Thanks!

**Comment by jacobsalmela** [comment]
I think you're asking about adding it to the other panes like Camera, Screen, etc. but it tccutil only touches the Accessibility portion.

https://github.com/jacobsalmela/tccutil/issues/52#issuecomment-940577858

**Comment by Torrekie** [comment]
Full list of available service strings (as of macOS 11.2.3)
```
kTCCServiceAccessibility
kTCCServiceAddressBook
kTCCServiceAll
kTCCServiceAppleEvents
kTCCServiceBluetoothAlways
kTCCServiceCalendar
kTCCServiceCamera
kTCCServiceContactsFull
kTCCServiceContactsLimited
kTCCServiceDeveloperTool
kTCCServiceFacebook
kTCCServiceFileProviderDomain
kTCCServiceFileProviderPresence
kTCCServiceLinkedIn
kTCCServiceListenEvent
kTCCServiceLiverpool
kTCCServiceLocation
kTCCServiceMediaLibrary
kTCCServiceMicrophone
kTCCServiceMotion
kTCCServicePhotos
kTCCServicePhotosAdd
kTCCServicePostEvent
kTCCServicePrototype3Rights
kTCCServicePrototype4Rights
kTCCServiceReminders
kTCCServiceScreenCapture
kTCCServiceShareKit
kTCCServiceSinaWeibo
kTCCServiceSiri
kTCCServiceSpeechRecognition
kTCCServiceSystemPolicyAllFiles
kTCCServiceSystemPolicyDesktopFolder
kTCCServiceSystemPolicyDeveloperFiles
kTCCServiceSystemPolicyDocumentsFolder
kTCCServiceSystemPolicyDownloadsFolder
kTCCServiceSystemPolicyNetworkVolumes
kTCCServiceSystemPolicyRemovableVolumes
kTCCServiceSystemPolicySysAdminFiles
kTCCServiceTencentWeibo
kTCCServiceTwitter
kTCCServiceUbiquity
kTCCServiceUserTracking
kTCCServiceWebKitIntelligentTrackingPrevention
kTCCServiceWillow
```
You can get this by extracting dyld shared cache and do `strings ${EXTRACTED_DSC}/System/Library/PrivateFrameworks/TCC.framework/Versions/A/TCC | grep kTCCService`

For camera and microphone accesses, you might have to edit tccutil.py to locate db path to your User's one, that will be `${HOME}/Library/Application Support/com.apple.TCC/TCC.db`

```python
# tccutil.py:32
# Database Path
tcc_db = '/Users/torrekie/Library/Application Support/com.apple.TCC/TCC.db'
```
As my case, Google Chrome was not requesting Camera and Microphone but complaining no permission.

Then do
```bash
sudo tccutil -i com.google.Chrome -s kTCCServiceCamera
sudo tccutil -i com.google.Chrome -s kTCCServiceMicrophone
```

**Comment by jacobsalmela** [comment]
Closing per https://github.com/jacobsalmela/tccutil/issues/52#issuecomment-940577858
